submariner/OVN-Interconnect.md

```diff
@@ -0,0 +1,231 @@
+  zones.
+
+```bash
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.3/16"]'
+      k8s.ovn.org/ovn-zone: global
+    name: cluster1-worker
+
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.5/16"]'
+      k8s.ovn.org/ovn-zone: az2
+    name: cluster2-worker
+```
+
+With the current architecture, Submariner adds routes only in the zone in which it is deployed. For example, if Submariner is deployed in
+zone 1 it programs OVN db in zone 1. So only pods in zone 1 nodes will be able to talk to other clusters. Pods in zone 2 or zone 3 will not
+be able to reach remote clusters connected via Submariner.
+
+### Route APIs
+
+As part of this proposal, we are planning to add two new CRDs in Submariner
+
+#### SubmarinerGwRoute
+
+This CR will be created when a remote endpoint is added and there will be one CR per remote cluster. This crd has two fields
+
+* NextHop - Specifies the next hop to reach the remote cluster.
+* RemoteCIDR - Specifies the list of  remote CIDRs reachable via this cluster, in this case it will be the IP of ovn-k8s-mp0
+interface.
+
+This CR will be used by the gateway pod to program OVN to send the traffic destined to remote clusters to the Submariner tunnel.
+
+``` go
+type SubmarinerGwRouteList struct {
+  metav1.TypeMeta `json:",inline"`
+  metav1.ListMeta `json:"metadata"`
+
+  Items []SubmarinerGwRoute `json:"items"`
+  }
+
+type SubmarinerRoutePolicySpec struct {
+    //Specifies the next hop to reach the remote CIDRs
+    NextHop string `json:"nextHop"`
+
+    //Specifies the remote CIDRs available via the next hop
+    RemoteCidr []string `json:"remoteCidr"`
+}
+```
+
+#### SubmarinerNonGWRoute
+
+This CR will be created when a remote cluster is connected. When more remote endpoints are added the Remote CIDR list will be updated.
+
+* NextHop - Specifies the next hop.
+* RemoteCIDR - Specifies the list of remote CIDRs reachable via this gateway.
+
+This CR will be used by the Routeagent pods to
+
+* In non-g/w node - send the traffic to the g/w node , if the current node is in a different zone
+* In g/w node - add route to send the traffic from other zones, destined to remote cluster to the submariner tunnel
+
+``` go
+type SubmarinerNonGWRouteList struct {
+    metav1.TypeMeta `json:",inline"`
+    metav1.ListMeta `json:"metadata"`
+
+    Items []SubmarinerNonGWRoute `json:"items"`
+}
+
+type SubmarinerRoutePolicySpec struct {
+    //Specifies the next hop to reach the remote CIDRs
+    NextHop string `json:"nextHop"`
+
+    //Specifies the remote CIDRs available via the next hop
+    RemoteCidr []string `json:"remoteCidr"`
+}
+```
+
+## Design Details
+
+In this proposal, the plan is to remove the current `submariner-router` and switches that were added to the OVN db. The network-plugin-syncer
+shall be removed  and can be replaced by controllers in Submariner Gateway and Submariner RouteAgent. Since we have multiple ovn db to
```

Comment:
We decided to use Route-agent pods and not the Submariner Gateway right?

Reply:
Done. I hope I did it all the places now.